### ISSUE TITLE
Conv filter visualization in Keras: Why do some filters get stuck at loss==0 ?

### ISSUE THREAD

**Issue by vrunm** [keras-team/keras#18334]
I am using this script from the Keras-Team to visualize conv filters of the VGG16 model:

https://github.com/keras-team/keras/blob/master/examples/conv_filter_visualization.py

For most filters, this works. However, if I try to generate the image for filter 11 in layer "block5_conv1", the algorithm gives me no output, because "some filters get stuck to 0" (line 156 ff):
```
# some filters get stuck to 0, we can skip them
if loss_value <= K.epsilon():
    return None

```
This is the only thing I changed in the script (last line):
`visualize_layer(vgg, "block5_conv1", output_dim=(112, 112), filter_range=(11, 12))`

However, for many other filters, e.g. block5_conv3, filter 1, I had no luck:
`visualize_layer(vgg, block5_conv3, output_dim=(112, 112), filter_range=(1, 2))`

also changed the output_dim, step, epochs, upscaling_steps, upscaling_factor but wasn't able to produce an image.

Is there a way, to generate a visualization of each filter reliably (based on the script provided)?

**Comment by tilakrayal** [comment]
@vrunm,
The above-mentioned link, which was provided for the code, was not working. Could you please provide the complete standalone code and the TensorFlow version you are using? Thank you!
